spark/v3.2/spark-extensions/src/test/java/org/apache/iceberg/spark/extensions/TestExpireSnapshotsProcedure.java

```diff
@@ -224,4 +235,62 @@ public void testConcurrentExpireSnapshotsWithInvalidInput() {
             catalogName, tableIdent, -1));
 
   }
+
+  @Test
+  public void testExpireDeleteFiles() throws Exception {
+    sql("CREATE TABLE %s (id bigint NOT NULL, data string) USING iceberg TBLPROPERTIES" +
+        "('format-version'='2', 'write.delete.mode'='merge-on-read')", tableName);
+
+    sql("INSERT INTO TABLE %s VALUES (1, 'a'), (2, 'b'), (3, 'c'), (4, 'd')", tableName);
+    sql("DELETE FROM %s WHERE id=1", tableName);
+
+    Table table = validationCatalog.loadTable(tableIdent);
+    table.refresh();
+
+    Assert.assertEquals("Should have 1 delete manifest", 1, deleteManifests(table).size());
+    Assert.assertEquals("Should have 1 delete file", 1, deleteFiles(table).size());
+    Path deleteManifestPath = new Path(deleteManifests(table).iterator().next().path());
+    Path deleteFilePath = new Path(String.valueOf(deleteFiles(table).iterator().next().path()));
+
+    sql("CALL %s.system.rewrite_data_files(table => '%s', options => map" +
+            "('delete-file-threshold','1', 'use-starting-sequence-number', 'false'))",
+        catalogName, tableIdent);
+
+    table.refresh();
+
+    sql("INSERT INTO TABLE %s VALUES (5, 'e')", tableName); // this txn moves the file to the DELETED state
+    sql("INSERT INTO TABLE %s VALUES (6, 'f')", tableName); // this txn removes the file reference
+
+    table.refresh();
```

Comment:
This refresh you would need if using the validation catalog

Reply:
Hm, I removed this refresh and it did not work (using Spark3Util to load the table in the beginning), not sure if I understood this right.

Reply:
Hmm that's odd to me, I thought that gave you the same catalog instance that spark has which should be attached to the same cache ... 🤷

Reply:
Oh well, keep what works